### PR TITLE
Add Atlas to jedi-stack

### PR DESCRIPTION
**Pull request by aerorahul**
This PR adds an option to build ECMWF::atlas with the stack.
Currently building v0.19.1

@mmiesch I saw you had a branch, but rather that using that, I just copied `build_fckit.sh` to `build_atlas.sh` and made the necessary changes.
This version of build does not use all the bells and whistles that Atlas can use e.g. FFTW, CGAL, etc.
But, it works for the limited use we have, so far.

### CODE REVIEW

**Comment by mmiesch** [comment]
Thanks @aerorahul - Yes, I had a branch but I haven't pursued it because at this point I think it is more practical to build it as part of the bundle.  To install atlas, you'd have to also install fckit, and we change both of these enough that the module+container maintenance would be significant.    The other reason I have held off on including atlas in the stack is because of functionality.  You mention that the ecmwf version "without all the bells and whistles" is sufficient for us now, but that is not the case if and when we merge in @benjaminmenetrier 's active pull requests in oops, saber, and fv3-jedi.   The problem is that we plan to make heavy use of unstructured grids.  The only option atlas currently has to create a mesh for an unstructured grid is Delauney triangulation.  But, in order to use this, you have to install and enable CGAL.  The problem there, as explained in https://github.com/JCSDA/jedi-stack/pull/56, is that this, as far as I can tell, re-introduces boost as a dependency.   CGAL itself is header-only now (as of v. 5.0) but I was unable to install it unless I first installed the ```boost_thread``` and ```boost_system``` libraries.

@benjaminmenetrier introduced a way to get around that with a ```no_connectivity``` mesh option as an alternative for unstructured grids.  But, for this you need to use the JCSDA fork.

So, in summary, I'm ok with adding this capability but I'm not sure we'll use it much until our development of fckit and atlas settles down a bit

File: modulefiles/mpi/compilerName/compilerVersion/mpiName/mpiVersion/atlas/atlas.lua [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+local hierA        = hierarchyA(pkgNameVer,2)
+local mpiNameVer   = hierA[1]
+local compNameVer  = hierA[2]
+local mpiNameVerD  = mpiNameVer:gsub("/","-")
+local compNameVerD = compNameVer:gsub("/","-")
+
+conflict(pkgName)
+
+always_load("eckit")
+always_load("fckit")
```
Comment (mmiesch) [inline modulefiles/mpi/compilerName/compilerVersion/mpiName/mpiVersion/atlas/atlas.lua]: It's probably not a problem for eckit and fckit, but I have moved away from using ```always_load```.  I have run into problems, particularly on Cheyenne, getting mixed up between the jedi-stack modules and the native modules - that is, it will load a module that's not compatible with the rest of the build.  But, again, this is probably not an issue with eckit and fckit because there are not likely to be native modules to cause confusion.

Comment (aerorahul) [reply modulefiles/mpi/compilerName/compilerVersion/mpiName/mpiVersion/atlas/atlas.lua]: > Thanks @aerorahul - Yes, I had a branch but I haven't pursued it because at this point I think it is more practical to build it as part of the bundle. To install atlas, you'd have to also install fckit, and we change both of these enough that the module+container maintenance would be significant. The other reason I have held off on including atlas in the stack is because of functionality. You mention that the ecmwf version "without all the bells and whistles" is sufficient for us now, but that is not the case if and when we merge in @benjaminmenetrier 's active pull requests in oops, saber, and fv3-jedi. The problem is that we plan to make heavy use of unstructured grids. The only option atlas currently has to create a mesh for an unstructured grid is Delauney triangulation. But, in order to use this, you have to install and enable CGAL. The problem there, as explained in #56, is that this, as far as I can tell, re-introduces boost as a dependency. CGAL itself is header-only now (as of v. 5.0) but I was unable to install it unless I first installed the `boost_thread` and `boost_system` libraries.
> 
> @benjaminmenetrier introduced a way to get around that with a `no_connectivity` mesh option as an alternative for unstructured grids. But, for this you need to use the JCSDA fork.
> 
> So, in summary, I'm ok with adding this capability but I'm not sure we'll use it much until our development of fckit and atlas settles down a bit

@mmiesch I am fine with keeping this as a place-holder for now.  This works for the work I am doing with Atlas.  This can be refactored/scrapped/enhanced when the development settles down. I leave it to you for an executive decision.

**Comment by aerorahul** [comment]
@mmiesch If there are no further comments, I would like to merge this by the end of the day today.
